File: ProjectJava/src/main/java/Domain/Donation.java
package Domain;

public class Donation extends Entity<Integer>{
    private CharityCase charityCase;
    private Donor donor;

    public Donation(CharityCase charityCase, Donor donor) {
        this.charityCase = charityCase;
        this.donor = donor;
    }

    public CharityCase getCharityCase() {
        return charityCase;
    }

    public void setCharityCase(CharityCase charityCase) {
        this.charityCase = charityCase;
    }

    public Donor getDonor() {
        return donor;
    }

    public void setDonor(Donor donor) {
        this.donor = donor;
    }

    @Override
    public String toString() {
        return "Donation{" +
                "charityCase=" + charityCase +
                ", donor=" + donor +
                '}';
    }
}

File: ProjectC#/WindowsFormsApp/WindowsFormsApp/Domain/Donation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp.Domain
{
    public class Donation:Entity<int>
    {
        private CharityCase charityCase;
        private Donor donor;

        public Donation(CharityCase charityCase, Donor donor)
        {
            this.charityCase = charityCase;
            this.donor = donor;
        }

        public CharityCase getCharityCase()
        {
            return charityCase;
        }

        public void setCharityCase(CharityCase charityCase)
        {
            this.charityCase = charityCase;
        }

        public Donor getDonor()
        {
            return donor;
        }

        public void setDonor(Donor donor)
        {
            this.donor = donor;
        }
    }
}
